app/controllers/storage_locations_controller.rb

```diff
@@ -75,6 +82,23 @@ def update
     end
   end
 
+  def deactivate
+    @storage_location = current_organization.storage_locations.kept.find(params[:storage_location_id])
+    @storage_location.discard!
+    redirect_to storage_locations_path, notice: "Storage Location deactivated successfully"
+  rescue Errors::StorageLocationNotEmpty => e
+    redirect_back(fallback_location: storage_locations_path(organization_id: current_organization), error: e.message)
+  end
+
+  def reactivate
+    @storage_location = current_organization.storage_locations.all.find(params[:storage_location_id])
```

Comment:
```suggestion
    @storage_location = current_organization.storage_locations.find(params[:storage_location_id])
```
I don't think you need to specify `all`